### PR TITLE
feat: Clarify expectations for client reports

### PR DESCRIPTION
**Pull request by philipphofmann**
This PR clarifies the expectations for client reports.
It is not required that the numbers for these are 100%
accurate and therefore the implemenation can loose some
data. Communicating this clearly will stop developers
implementing a more complex solution which would ensure
to never loose any client reports.

### CODE REVIEW

File: src/docs/sdk/client-reports.mdx [hunk 1]
```diff
@@ -98,14 +98,23 @@ need to match the reason codes that relay would emit directly to Sentry.
...
+the SDKs should ensure not dropping too many reports. It is not required, for example: 
+ - to persist the data when an application crashes.
+ - to move an envelope item with a client report to the next envelope when the cache for envelopes is full. 
+
+SDKs are encouraged to reduce the total amount of needless communication, especially
+on mobile. A recommended approach is to keep track of the discard events in the
+transport directly and attach it to an already scheduled envelope or periodically flush
+them out as separate envelope items.
+For mobile SDKs, the recommendation is to attach the client report to the next envelope
+enqueued for submission instead of periodically flushing them out. We accept the tradeoff
```
Comment (marandaneto) [inline src/docs/sdk/client-reports.mdx:113]: This part looks a bit weird, we mention that SDKs are encouraged to reduce the total of communication, but then we say: `or periodically flush
them out as separate envelope items.`

Comment (philipphofmann) [reply src/docs/sdk/client-reports.mdx:113]: I didn't write that initially. I guess the intention of @mitsuhiko was that you shouldn't send an envelope item each time an event is dropped. Especially on the backend you could have several items dropped per second and doing an HTTP request for each of them would be crazy. I will try to rephrase that a bit.

Comment (philipphofmann) [reply src/docs/sdk/client-reports.mdx]: Is it better now @marandaneto?

**Review by mitsuhiko — APPROVED** [review]

File: src/docs/sdk/client-reports.mdx (hunk 1 above)
Comment (marandaneto) [reply src/docs/sdk/client-reports.mdx]: yes, thanks.

**Pull request merged**